### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Mahamadou-Nouridine**
In this Pr, I brought changes according to the following requirements:

- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (tufoinnkuo10) [inline app/models/user.rb]: - [x] Great job so far. Please kindly ensure  `:confirmable ` is also enabled in the user model. This enables the application to asks for confirmation of email. it is important as it verifies if an account is already confirmed to sign in, and to send emails with confirmation instructions. please Don't forget to activate it in the devise migration file as well.
- [ ] please remove the generated comments in the user model for cleaner code practices. 😄

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @team

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ devise well added

Required Changes ♻️
Check the comments under the review

la

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by Munsa1 — APPROVED** [review]
Hi @Mahamadou-Nouridine, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Mahamadou-Nouridine** [comment]
For my last reviewer: I have done the require changes, but I don't know for which reason it went above the first review [(commit)](https://github.com/Mahamadou-Nouridine/blog-app/pull/9/commits/64aa53c771fd319dce4a1199970b5e0320a6dbad). Kindly, consider checking the [files changed](https://github.com/Mahamadou-Nouridine/blog-app/pull/9/files) tab

**Comment by Munsa1** [comment]
> For my last reviewer: I have done the require changes, but I don't know for which reason it went above the first review [(commit)](https://github.com/Mahamadou-Nouridine/blog-app/pull/9/commits/64aa53c771fd319dce4a1199970b5e0320a6dbad). Kindly, consider checking the [files changed](https://github.com/Mahamadou-Nouridine/blog-app/pull/9/files) tab

Okay noted @Mahamadou-Nouridine

**Pull request merged**